dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/eventdatavalue/DefaultEventDataValueService.java

```diff
@@ -116,16 +119,32 @@ public void processDataValues( ProgramStageInstance programStageInstance, Event
             {
                 // This can happen if a wrong data element identifier is provided
                 importSummary.getConflicts().add( new ImportConflict( "dataElement", dataValue.getDataElement() + " is not a valid data element" ) );
+                continue;
             }
-            else if ( validateDataValue( programStageInstance, importOptions.getUser(), dataElement, dataValue.getValue(), importSummary )
-                && !importOptions.isDryRun() )
+
+            if ( validateDataValue( programStageInstance, importOptions.getUser(), dataElement, dataValue.getValue(), importSummary ) )
             {
+                accessibleDataElements.add( dataValue.getDataElement() );
                 prepareDataValueForStorage( dataElementValueMap, dataValue, dataElement, newDataValues, updatedDataValues,
                     removedDataValuesDueToEmptyValue, storedBy );
             }
+            else
+            {
+                nonAccessibleDataElements.add( dataValue.getDataElement() );
+            }
         }
 
-        programStageInstanceService.auditDataValuesChangesAndHandleFileDataValues( newDataValues, updatedDataValues, removedDataValuesDueToEmptyValue, dataElementsCache, programStageInstance, singleValue );
+        Set<String> checkedDataElements = Sets.union( accessibleDataElements, nonAccessibleDataElements );
+
+        for ( DataElement dataElement : programStageInstance.getProgramStage().getDataElements() )
+        {
+            if ( !checkedDataElements.contains( dataElement.getUid() ) &&  !trackerAccessManager.canWrite( importOptions.getUser(), programStageInstance, dataElement, true ).isEmpty() )
+            {
+                nonAccessibleDataElements.add( dataElement.getUid() );
+            }
+        }
+
+        programStageInstanceService.auditDataValuesChangesAndHandleFileDataValues( newDataValues, updatedDataValues, removedDataValuesDueToEmptyValue, dataElementsCache, nonAccessibleDataElements, programStageInstance, singleValue );
```

Comment:
minor: can you reformat to respect 80 chars length

```
programStageInstanceService.auditDataValuesChangesAndHandleFileDataValues( newDataValues, updatedDataValues,
    removedDataValuesDueToEmptyValue, dataElementsCache, nonAccessibleDataElements, programStageInstance,
    singleValue )
```

Reply:
done.